### PR TITLE
KK-29 | Submit registration form data as mutation query

### PR DESCRIPTION
**Pull request by badrange**

### CODE REVIEW

File: src/domain/registration/form/RegistrationForm.tsx [hunk 1]
```diff
@@ -1,197 +1,207 @@
...
+  initialValues,
+}) => {
+  const handleSubmit = (values: RegistrationFormValues) => {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    //const submitValues: SubmitChildValues = convertFormValuesToSubmitChildMutation(
+    //  values
+    //);
   };
+  const [submitChild, { data }] = useMutation(submitChildMutationQuery);
+  console.log(data);
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationForm.tsx]: Remove console.log

File: src/domain/registration/form/RegistrationForm.tsx [hunk 2]
```diff
@@ -1,197 +1,207 @@
...
-    const { setFormValues } = this.props;
-
-    setFormValues(values);
+const RegistrationForm: FunctionComponent<Props> = ({
+  setFormValues,
+  initialValues,
+}) => {
+  const handleSubmit = (values: RegistrationFormValues) => {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    //const submitValues: SubmitChildValues = convertFormValuesToSubmitChildMutation(
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationForm.tsx]: Remove all those comment also

File: src/domain/registration/form/RegistrationForm.tsx [hunk 3]
```diff
@@ -1,197 +1,207 @@
...
-                  <div className={styles.childInfo}>
+  return (
+    <div className={styles.registrationForm}>
+      <Translation>
+        {t => (
+          <Formik
+            initialValues={initialValues}
+            onSubmit={e => {
+              setFormValues(e);
+              //e.preventDefault();
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationForm.tsx]: that comment line looks valid, should be first before setFormValues

File: src/domain/registration/form/RegistrationFormUtils.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  RegistrationFormValues,
+  SubmitChildValues,
+} from '../types/RegistrationTypes';
+
+/**
+ * Convert FormValues from RegistrationForm to childMutation
+ * @param {RegistrationFormValues} stateFormValues Values from state.
+ * @returns {HomeFormValues} initialValues used in form.
+ */
+export const convertFormValuesToSubmitChildMutation = (
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationFormUtils.ts]: Why we even need this? Data in redux store is already beautified for sending

File: src/domain/registration/mutations/submitChild.ts [hunk 5]
```diff
@@ -1,6 +1,17 @@
+import { useMutation } from '@apollo/react-hooks';
 import { gql } from 'apollo-boost';
 
-const submitChild = gql`
+import { SubmitChildValues } from '../types/RegistrationTypes';
+import { submitChild } from '../../api/generatedTypes/submitChild';
+/*
+import {
+  submitChild,
+  submitChild_submitChild,
```
Comment (midori0507) [inline src/domain/registration/mutations/submitChild.ts]: No comment section, delete, or leave reason to comment out

File: src/domain/registration/form/RegistrationFormUtils.ts (hunk 4 above)
Comment (midori0507) [reply src/domain/registration/form/RegistrationFormUtils.ts]: is it backend var required?

Comment (badrange) [reply src/domain/registration/form/RegistrationFormUtils.ts]: True! Deleted!

File: src/domain/api/client.ts [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
+
+function select(state: StoreState) {
+  return state.authentication.backend.apiToken;
+}
 
 export default new ApolloClient({
   request: async operation => {
     try {
-      const user = await getAuthenticatedUser();
+      const token = select(store.getState());
```
Comment (midori0507) [inline src/domain/api/client.ts]: That looks strange.. how could store state is the token..

File: src/domain/registration/form/RegistrationForm.tsx [hunk 7]
```diff
@@ -1,197 +1,199 @@
...
+                firstName: e.child.firstName,
+                lastName: e.child.lastName,
+                guardianFirstName: e.guardian.firstName,
+                guardianLastName: e.guardian.lastName,
+                email: e.guardian.email,
+              },
+            });
+          } catch (err) {
+            // TODO: Error handling.
+            console.error(err);
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationForm.tsx]: Put a eslint-ignore here so it stop give you warning

**Pull request merged**